### ISSUE TITLE
Error in core test_get_component_classes_in_namespace in Moodle 4.2 after #68

### ISSUE THREAD

**Issue by pmaneggia** [mudrd8mz/moodle-mod_subcourse#75]
The fix https://github.com/catalyst/moodle-mod_subcourse/issues/68 (surely far more important than this one) causes an error in core `component_test::test_get_component_classes_in_namespace`.

`coding_exception: Coding error detected, it must be fixed by a programmer: When including this file for a unit test, the test must be run in an isolated process. See the PHPUnit @runInSeparateProcess and @runTestsInSeparateProcesses annotations.`

I guess it will be fixed by removing the require_once and use the new external API functions in a new branch MOODLE_402_STABLE (see https://github.com/mebis-lp/moodle-mod_subcourse/tree/MBS-8535-fix-core-test_get_component_classes_in_namespace).

**Comment by danmarsden** [comment]
thanks for the report! - feel free to fire a PR through with the fix for this against the main 4.1 branch, just include in the comments that it needs to be applied to a 4.2 and higher branch only and I'll create a specific 4.2 branch so we can merge it in.

I did notice another change in your patch related to the form but I didn't look closely to see what that was for - if it's a separate issue we should do that as a separate PR covering the reason it's being done - thanks!

**Comment by pmaneggia** [comment]
Hi,

happily: https://github.com/catalyst/moodle-mod_subcourse/pull/76
Is the comment adequate?

The other change you saw was a previous unrelated commit in our fork, I had not brought the branch in shape for a pull request yet.
